### PR TITLE
oh-my-zsh: add module

### PR DESCRIPTION
**Pull request by uvNikita**
Again, I basically reused implementation from the NixOs [oh-my-zsh](https://github.com/NixOS/nixpkgs/blob/release-17.03/nixos/modules/programs/zsh/oh-my-zsh.nix) module.

### CODE REVIEW

**Comment by uvNikita** [comment]
Example configuration:

```
programs = {
    zsh = {
      enable = true;
      enableCompletion = true;
      enableAutosuggestions = true;
      historySize = 20000;
      oh-my-zsh = {
        enable = true;
        theme = "robbyrussell";
        plugins = [
          "git" "_docker" "systemd" "themes" "copyfile"
          "fabric" "gitignore" "sudo" "pass" "command-not-found"
        ];
      };
    };
  };
};
```

File: modules/programs/oh-my-zsh.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+let
+
+  cfg = config.programs.zsh.oh-my-zsh;
+
+in
+
+{
+  options = {
+    programs.zsh.oh-my-zsh = {
+      enable = mkOption {
```
Comment (rycee) [inline modules/programs/oh-my-zsh.nix]: I think `mkEnableOption` is preferrable.

File: modules/programs/oh-my-zsh.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        default = false;
+        description = ''
+          Enable oh-my-zsh.
+        '';
+      };
+
+      plugins = mkOption {
+        default = [];
+        type = types.listOf(types.str);
```
Comment (rycee) [inline modules/programs/oh-my-zsh.nix]: Can add example?

File: modules/programs/oh-my-zsh.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        default = "";
+        type = types.str;
+        description = ''
+          Path to a custom oh-my-zsh package to override config of oh-my-zsh.
+        '';
+      };
+
+      theme = mkOption {
+        default = "";
+        type = types.str;
```
Comment (rycee) [inline modules/programs/oh-my-zsh.nix]: Can add example?

File: modules/programs/oh-my-zsh.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.oh-my-zsh ];
+    programs.zsh.initExtra = with pkgs; ''
+      # oh-my-zsh configuration generated by NixOS
+      export ZSH=${oh-my-zsh}/share/oh-my-zsh
+      ${optionalString (length(cfg.plugins) > 0)
+        "plugins=(${concatStringsSep " " cfg.plugins})"
+      }
+      ${optionalString (stringLength(cfg.custom) > 0)
```
Comment (rycee) [inline modules/programs/oh-my-zsh.nix]: Better with just `cfg.custom != ""`, same with `cfg.theme` below.

File: modules/programs/oh-my-zsh.nix [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.oh-my-zsh ];
+    programs.zsh.initExtra = with pkgs; ''
+      # oh-my-zsh configuration generated by NixOS
+      export ZSH=${oh-my-zsh}/share/oh-my-zsh
+      ${optionalString (length(cfg.plugins) > 0)
```
Comment (rycee) [inline modules/programs/oh-my-zsh.nix]: `cfg.plugins != []`

File: modules/programs/oh-my-zsh.nix [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        default = [];
+        type = types.listOf(types.str);
+        description = ''
+          List of oh-my-zsh plugins
+        '';
+      };
+
+      custom = mkOption {
+        default = "";
+        type = types.str;
```
Comment (rycee) [inline modules/programs/oh-my-zsh.nix]: May example? Not sure what would be a good one here, though.

**Comment by rycee** [comment]
Rebased to master in 3ef56576d35d2ba245295443c4998cb51ea3e87b. Many thanks!

**Pull request closed without merging**